### PR TITLE
[DD4hep]  use DD4hep consistently in configurations

### PR DESCRIPTION
**Pull request by ianna**
#### PR description:

address issue #33373

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

**Comment by ianna** [comment]
please test

**Comment by tvami** [comment]
+1
 * tests pass
 * please note the the ES get() migration still needs to be done here, see https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7e7a6c/20273/llvm-analysis/esrget-sa.txt

**Comment by srimanob** [comment]
+Upgrade

Technical PR to resolve the issue of mixing between DD4hep and else (DD4Hep,...). No change is expected.

**Comment by civanch** [comment]
+1

**Comment by perrotta** [comment]
+1

File: Geometry/EcalAlgo/plugins/BuildFile.xml [hunk 1]
```diff
@@ -7,6 +7,6 @@
   <flags EDM_PLUGIN="1"/>
 </library>
 
-<library file="Ecal*.cc moduleDD4Hep.cc" name="GeometryEcalAlgoDD4HepPlugin">
+<library file="Ecal*.cc moduleDD4hep.cc" name="GeometryEcalAlgoDD4hepPlugin">
```
Comment (cvuosalo) [inline Geometry/EcalAlgo/plugins/BuildFile.xml:10]: Should the file `moduleDD4Hep.cc` in `Geometry/EcalAlgo/plugins` have its name changed to `moduleDD4hep.cc`?

**Comment by cvuosalo** [comment]
The file `Geometry/CSCGeometryBuilder/test/python/validateCSCGeometryDD4hep_cfg.py` contains the string `validateCSCGeometryDD4HEP.root`. Should it be changed for consistency?

File: Geometry/HcalTestBeamData/interface/HcalTB02ParametersFromDD.h [hunk 2]
```diff
@@ -18,7 +18,7 @@ class HcalTB02ParametersFromDD {
 
 private:
   static constexpr double k_ScaleFromDDDToG4 = 1.0;
-  static constexpr double k_ScaleFromDD4HepToG4 = 10.0;
+  static constexpr double k_ScaleFromDD4hepToG4 = 10.0;
```
Comment (cvuosalo) [inline Geometry/HcalTestBeamData/interface/HcalTB02ParametersFromDD.h:21]: @bsunanda Not related to this PR, but this value looks wrong. This file may have been missed in the units conversion. I think it should be:
```
 static constexpr double k_ScaleFromDD4hepToG4 = (1.0 / dd4hep::mm);
```

**Comment by cvuosalo** [comment]
Doing `git grep DD4Hep` with this PR reveals about 20 files that still contain `DD4Hep`.

File: Geometry/HcalTestBeamData/interface/HcalTB02ParametersFromDD.h (hunk 2 above)
Comment (bsunanda) [reply Geometry/HcalTestBeamData/interface/HcalTB02ParametersFromDD.h:21]: OK Carl I shall take care of this

**Comment by perrotta** [comment]
> Doing `git grep DD4Hep` with this PR reveals about 20 files that still contain `DD4Hep`.

Moreover: with lxr I still can find 53 files with "DD4Hep" in the name:
https://cmssdt.cern.ch/lxr/search?%21v=CMSSW_12_2_X_2021-11-05-2300&_filestring=DD4Hep&_string=&_casesensitive=1
and 66 occurrencies from 33 different files with "DD4Hep" in the body:
https://cmssdt.cern.ch/lxr/search?%21v=CMSSW_12_2_X_2021-11-05-2300&_filestring=DD4Hep&_string=&_casesensitive=1

Noteworthy, there are also two files in \test with "DD4HEP" in full capital letters in their names:
https://cmssdt.cern.ch/lxr/search?%21v=CMSSW_12_2_X_2021-11-05-2300&_filestring=DD4HE&_string=&_casesensitive=1
but I think they can stay as such, those names following some reasonable convention

**Comment by ianna** [comment]
> Doing `git grep DD4Hep` with this PR reveals about 20 files that still contain `DD4Hep`.

@cvuosalo - thanks! I'll have another go at CMSSW now that it is merged. There were way too many hits before. Thanks!

**Comment by ianna** [comment]
There are around 70 places where `DD4Hep` is mentioned: https://cmssdt.cern.ch/lxr/search?%21v=CMSSW_12_2_X_2021-11-05-2300&_filestring=&_string=DD4Hep&_casesensitive=1

File: Geometry/EcalAlgo/plugins/BuildFile.xml (hunk 1 above)
Comment (ianna) [reply Geometry/EcalAlgo/plugins/BuildFile.xml:10]: done in https://github.com/cms-sw/cmssw/pull/36021

**Comment by ianna** [comment]
@cvuosalo and @perrotta - I hope, all your comments are addressed. Please, see https://github.com/cms-sw/cmssw/pull/36021

**Pull request merged**